modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -25,35 +25,77 @@ import {
 	YAxis,
 } from 'recharts';
 
-const BAR_CHART = {
-	barHeight: 16,
-	dotRadiusMax: 35,
-	dotRadiusMin: 5,
-	fillXAxis: '#666',
-	height: 400,
-	legendHeight: 70,
-	stroke: '#E7E7ED',
-	width: 1150,
-};
+import {BAR_CHART, COLORS} from '../utils/constants';
+
+export default function AuditBarChart({categories, rtl}) {
+	const auditBarChartData = useMemo(() => {
+		const dataKeys = new Set();
+
+		const bars = categories.reduce((acc, category) => {
+			if (!category.children) {
+				return undefined;
```

Comment:
Are you sure you want to return undefined? This means that if we get to like the 4th category, and if it doesn't have children, we now have an accumulator that is set to undefined, rather than its current value. I would expect this to be `return acc;`

Reply:
And if that happens, `acc.concat` below will 💥 on the next iteration in the loop (if there is one).